projects/Mallard/src/components/article/article-headline/index.tsx

```diff
@@ -13,17 +17,53 @@ const styles = StyleSheet.create({
         marginRight: metrics.horizontal * 2,
         marginTop: metrics.vertical / 2,
     },
+    highlightBg: {
+        ...StyleSheet.absoluteFillObject,
+        zIndex: -1,
+    },
 })
 
 const ArticleHeadline = ({
     children,
     textStyle,
     weight,
+    hasHighlight = false,
 }: ArticleHeadlineProps) => {
+    const [boxes, setBoxes] = useState<RectangleType[]>([])
+    const onTextLayout = (ev: any) => {
```

Comment:
My only tidiness thing would be to break this out into a named hook and have the magic numbers as default parameters with nice parameter names so it's clear what they do.

Reply:
good idea!